File: 2837-minimum-operations-to-make-the-integer-zero/solution.ts
function makeTheIntegerZero(num1: number, num2: number): number {
    for(let i = 1; i <= 61; i++) {
        num1 -= num2;
        if(num1 <= 0) return -1;
        const cnt = num1.toString(2).match(/[1]/g)?.length || 0;
        if(cnt <= i && num1 >= i) return i;
    }
    return -1;
};

File: 2837-minimum-operations-to-make-the-integer-zero/solution.cpp
class Solution {
public:
    int makeTheIntegerZero(int num1, int num2) {
        long long n1 = num1, n2 = num2;
        for (int i = 1; i <= 61; i++) {
            n1 -= n2;
            if (n1 <= 0) return -1;
            int cnt = __builtin_popcountll(n1); // count of '1's in binary
            if (cnt <= i && n1 >= i) return i;
        }
        return -1;
    }
};
